### PR TITLE
Update custom_training.ipynb custom training basics tutorial

### PR DESCRIPTION
**Pull request by 8bitmp3**
Some suggested (UX) improvements to the _'Custom training: basics'_ r2 (TensorFlow 2 Beta1) tutorial:
- Update and expand `tf.keras` paragraphs with a link to the Keras guide
- Expand the L2 loss section
- Expand the training data synthesis with Gaussian noise section
- Update the training model section
- Add a small explanation of `tf.GradientTape` and `tf.assign_sub`
- Add backticks (`tf.Variable`)
- Change Copyright to `2019`
- Use second person where applicable
- Add/replace commas/colons
- Use em dash (`—`)/replace en dash(`-`)
- Add a `tf.random.normal` reference
- Use % modulo op to print the 'Current loss' in line with the rest of the notebook's style
- Fix labels in a plot (`True W`, `True b`)
- Change "desired" to "target" (commonly used word for `y`)

As before, if you find any of it useful (or not) let me know @MarkDaoust @lamberta. Any feedback welcome. Cheers!

### CODE REVIEW

File: site/en/r2/tutorials/eager/custom_training.ipynb [hunk 1]
```diff
@@ -74,10 +74,10 @@
...
+        "In the previous tutorial, you covered the TensorFlow APIs for automatic differentiation—a basic building block for machine learning.\n",
+        "In this tutorial, you will use the TensorFlow primitives introduced in the prior tutorials to do some simple machine learning.\n",
         "\n",
-        "TensorFlow also includes a higher-level neural networks API (`tf.keras`) which provides useful abstractions to reduce boilerplate. We strongly recommend those higher level APIs for people working with neural networks. However, in this short tutorial we cover neural network training from first principles to establish a strong foundation. "
+        "TensorFlow also includes `tf.keras`—a high-level neural network API, which provides useful abstractions to reduce boilerplate and makes TensorFlow easier to use without sacrificing flexibility and performance. We strongly recommend [this high-level API](https://www.tensorflow.org/beta/guide/keras/overview) for people working with neural networks. However, in this short tutorial you will learn how to train a neural network from first principles to establish a strong foundation."
```
Comment (MarkDaoust) [inline site/en/r2/tutorials/eager/custom_training.ipynb]: https://www.tensorflow.org/beta/guide/keras/overview

Prefer relative links so you can stay in tensorflow.org/github/colab. Use:

\[this high-level API](../../guide/keras/overview.ipynb)

File: site/en/r2/tutorials/eager/custom_training.ipynb [hunk 2]
```diff
@@ -142,9 +142,9 @@
         "id": "wfneTXy7JcUz"
       },
       "source": [
-        "TensorFlow, however, has stateful operations built in, and these are often more pleasant to use than low-level Python representations of your state. To represent weights in a model, for example, it's often convenient and efficient to use TensorFlow variables.\n",
+        "TensorFlow, however, has stateful operations built in, and these are often more pleasant to use than low-level Python representations of your state. To represent weights in a model, for example, it's often convenient and efficient to use TensorFlow variables, which are manipulated via the `tf.Variable` class.\n",
```
Comment (MarkDaoust) [inline site/en/r2/tutorials/eager/custom_training.ipynb]: "it's often convenient and efficient to use TensorFlow variables, which are manipulated via the `tf.Variable` class"


How about just:

"To represent weights in a model use the `tf.Variable` class."

File: site/en/r2/tutorials/eager/custom_training.ipynb [hunk 3]
```diff
@@ -142,9 +142,9 @@
...
         "\n",
-        "A Variable is an object which stores a value and, when used in a TensorFlow computation, will implicitly read from this stored value. There are operations (`tf.assign_sub`, `tf.scatter_update`, etc) which manipulate the value stored in a TensorFlow variable."
+        "A variable is an object, which stores a value and, when used in a TensorFlow computation via `tf.Variable`, will implicitly read from this stored value. There are operations (`tf.assign_sub`, `tf.scatter_update`, etc), which manipulate the value stored in a TensorFlow variable."
```
Comment (MarkDaoust) [inline site/en/r2/tutorials/eager/custom_training.ipynb]: Same here:

"A tf.Variable` is an object, which stores a value, and will implicitly read from this stored value."

File: site/en/r2/tutorials/eager/custom_training.ipynb [hunk 4]
```diff
@@ -272,7 +273,7 @@
       "source": [
         "### Obtain training data\n",
         "\n",
-        "Let's synthesize the training data with some noise."
+        "First, you can synthesize the training data by adding random Gaussian (Normal) noise to the inputs, which can make your model more robust against overfitting and thus aid generalization:"
```
Comment (MarkDaoust) [inline site/en/r2/tutorials/eager/custom_training.ipynb]: Delete this line:

"which can make your model more robust against overfitting and thus aid generalization:"

This is not here for regularization, it is here to simulate noisy real-world data. They're different things and this will only confuse people.

File: site/en/r2/tutorials/eager/custom_training.ipynb (hunk 1 above)
Comment (8bitmp3) [reply site/en/r2/tutorials/eager/custom_training.ipynb]: Got it @MarkDaoust, thanks. Noticed these relative links after the PR elsewhere and was going to ask 👍

File: site/en/r2/tutorials/eager/custom_training.ipynb (hunk 2 above)
Comment (8bitmp3) [reply site/en/r2/tutorials/eager/custom_training.ipynb]: Agreed, @MarkDaoust. Done. I should watch the [style and tone](https://developers.google.com/style/tone).

File: site/en/r2/tutorials/eager/custom_training.ipynb (hunk 3 above)
Comment (8bitmp3) [reply site/en/r2/tutorials/eager/custom_training.ipynb]: Yup! It's simpler and avoids repetition ("...variable... `variable`...") @MarkDaoust . Done

File: site/en/r2/tutorials/eager/custom_training.ipynb (hunk 4 above)
Comment (8bitmp3) [reply site/en/r2/tutorials/eager/custom_training.ipynb]: Got it! Done. Thanks @MarkDaoust

**Comment by 8bitmp3** [comment]
Syntax error from the 1st commit fixed https://github.com/tensorflow/docs/pull/756/commits/a806039cb94bdf04f71b0c797ca964f531ee10e2 @MarkDaoust

**Comment by 8bitmp3** [comment]
just checking if anything else is outstanding on my side @MarkDaoust

**Comment by 8bitmp3** [comment]
thanks @lamberta 🙏

File: site/en/r2/tutorials/eager/custom_training.ipynb [hunk 5]
```diff
@@ -246,22 +263,22 @@
       "source": [
         "### Define a loss function\n",
         "\n",
-        "A loss function measures how well the output of a model for a given input matches the desired output. Let's use the standard L2 loss."
+        "A loss function measures how well the output of a model for a given input matches the target output. It is like an objective function, the quantity of which will be minimized during training. Let's use the standard L2 loss, also known as the least square errors:"
```
Comment (lamberta) [inline site/en/r2/tutorials/eager/custom_training.ipynb]: @MarkDaoust *objective function* isn't a term I've seen in the TF docs and not sure it clarifies here. wdyt?

**Comment by lamberta** [comment]
Thanks! Looks good and sorely needed on this one. A points Mark should look at, but good on my side :)

File: site/en/r2/tutorials/eager/custom_training.ipynb (hunk 5 above)
Comment (8bitmp3) [reply site/en/r2/tutorials/eager/custom_training.ipynb]: @MarkDaoust @lamberta from the theory I understand that loss function = error function = objective function (as mentioned in DeepMind's Andrew Trask's excellent Grokking Deep Learning book). Perhaps, keeping it simple is better: `"The goal is to minimize it during training"`. This is still useful for those folks who are new to ML imo.

Comment (lamberta) [reply site/en/r2/tutorials/eager/custom_training.ipynb]: Yeah, I like that idea. Let's remove the object function term and keep your explanation. Thanks
